### PR TITLE
feat(alertmanager): Add exporter factory and config

### PR DESCRIPTION
**Pull request by mcube8**
**Description:** Factory implementation of Alertmanager Exporter
Initial PR - base configs and factory implementation

**Link to tracking Issue:** [#23659](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23569)

**Testing:** Unit tests for config and factory implementation

**Documentation:** Readme and Sample Configs to use Alertmanager exporter

### CODE REVIEW

**Review by songy23 — COMMENTED** [review]
Please add a codeowner file, update builders and fix lint errors.

**Comment by sokoide** [comment]
@songy23, thank you for your suggestion. When we ran `make update-codeowners`, we got `codeowners are not members: BinaryFissionGames...`. Is it a known issue and will we need to wait for a resolution of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/26124? If there is a workaround, please kindly share. Thank you!

**Comment by songy23** [comment]
Seems OK to leave it out for now https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/26124#issuecomment-1696159037

> See https://github.com/open-telemetry/community/issues/1659 for a long term fix. It's ok to comment this out for now.

File: exporter/alertmanagerexporter/factory.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	"go.opentelemetry.io/collector/config/configopaque"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/alertmanagerexporter/internal/metadata"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr   = "alertmanager"
```
Comment (jpkrohling) [inline exporter/alertmanagerexporter/factory.go:22]: This is unused now, right?

**Review by jpkrohling — COMMENTED** [review]
Looks like a great start! Can you please go over the original comments I had on the other PR and see if everything is addressed? I had a comment about using "http://example.com/alert" as a default value, and this seems to still be present in this PR here. I'm see that other points were indeed addressed though, thank you for that!

File: exporter/alertmanagerexporter/factory.go (hunk 1 above)
Comment (mcube8) [reply exporter/alertmanagerexporter/factory.go]: I updated the default to "http://example.com/otelcol" and added a note in Readme. If you suggest a different default config value, kindly share.

File: exporter/alertmanagerexporter/README.md [hunk 2]
```diff
@@ -28,7 +28,7 @@ The following settings are optional:
 - `timeout` `sending_queue` and `retry_on_failure` settings as provided by [Exporter Helper](https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/exporterhelper#configuration)
 - [HTTP settings](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/confighttp/README.md)
 - [TLS and mTLS settings](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configtls/README.md)
-- `generator_url` is the source of the alerts to be used in Alertmanager's payload and can be set to the URL of the opentelemetry collector if required
+- `generator_url` is the source of the alerts to be used in Alertmanager's payload. The default value is <http://example.com/otelcol>, and can be set to the URL of the opentelemetry collector.
```
Comment (jpkrohling) [inline exporter/alertmanagerexporter/README.md:31]: I would rather set the default to `opentelemetry-collector` if an URL isn't required, or simply `<unset-otelcol>`. Using `example.com` seems out of place here.

Comment (mcube8) [reply exporter/alertmanagerexporter/README.md]: Thank you, I have updated the default to "opentelemetry-collector" now.

**Review by jpkrohling — APPROVED** [review]

**Comment by jpkrohling** [comment]
Can you please resolve the conflicts? Once that is done and the CI is green, I'll merge this PR.

**Comment by mcube8** [comment]
> Can you please resolve the conflicts? Once that is done and the CI is green, I'll merge this PR.

The merge conflict has now been resolved, thank you!

**Pull request merged**